# Title: Guided Project on Profitable App Profiles for the App Store and Google Play Markets

This project is about apps found on Play store and App Store, where the goal
is to find the best profitable apps on both the app market places.

In [1]:
def explore_data(dataset, start, end, rows_and_columns = False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') #adds a new (empty) line after each row
    if rows_and_columns:
        print('Number of rows: ', len(dataset))
        print('Number of columns', len(dataset[0]))

In [2]:
from csv import reader
with open('googleplaystore.csv') as f:
    read_file = reader(f, delimiter = ',')
    gplaydatacomp = list(read_file)
    
    gplaycol = gplaydatacomp[:1]
    gplaydata = gplaydatacomp[1:]

with open('AppleStore.csv') as f:
    red_file = reader(f, delimiter = ',')
    Applecomp = list(red_file)
    Applecol = Applecomp[:1]
    Appledata = Applecomp[1:]

In [3]:
print(gplaycol)
explore_data(gplaydata,0,3, True)


[['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']]
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows:  10841
Number of columns 13


We see that the Google Play data set has 10841 apps and 13 columns. At a quick glance, the columns that might be useful for the purpose of our analysis are 'App',  =='Category'== , 'Reviews', 'Installs', 'Type', 'Price', and > Genres.

Not all column names are self-explanatory in this case, but details about each column can be found in the data set [documentation.](https://www.kaggle.com/datasets/lava18/google-play-store-apps)

In [4]:
print(Applecol)
explore_data(Appledata, 0, 2, True)

[['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']]
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows:  7197
Number of columns 16


# Removing inaccurate data

The column below shows an inaccurate data in the dataset, we can remove this data found from the discussions forum to enhance correct analysis of the dataset.

In [5]:
print(gplaydata[10472])
print(len(gplaycol) == len(gplaydata[10472]))
print(gplaycol)
oldgplaydata = gplaydata
print(oldgplaydata[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
False
[['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']]
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [6]:
del gplaydata[10472] # An example that shows how to remove the duplicate instances of the list of lists.
print(gplaydata[10472])
print('\n')

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']




# To Check duplicate entries


To check duplicate entries in the Google play store app data

It is possible that the dataset may or may not contain duplicate values, to best rule out the possibility and prevent any miss in analysis further down the line. It would be effective to check duplicity in data prior to any analysis.

In [7]:
#To check duplicity in google play dataset

unique_ntris = []
duplicate_ntris = []

for row in gplaydata:
    if row[0] in unique_ntris:
        duplicate_ntris.append(row[0])
    else:
        unique_ntris.append(row[0])
        
print("Number of duplicate entries: ",len(duplicate_ntris))
print(duplicate_ntris[:4])

Number of duplicate entries:  1181
['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings']


In [8]:
#To check duplicity in app store dataset

unique_ntris_apl = []
duplicate_ntris_apl = []

for row in Appledata:
    if row[0] in unique_ntris_apl:
        duplicate_ntris_apl.append(row[0])
    else:
        unique_ntris_apl.append(row[0])
        
print("Number of duplicate entries: ",len(duplicate_ntris_apl))
print(duplicate_ntris_apl[:4])
print(Appledata[:5])

Number of duplicate entries:  0
[]
[['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1'], ['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1'], ['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1'], ['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']]


It seems that the Apple dataset does not have any duplicate entries. We will thus proceed to check if there are any app names with non-english characters in them.

In [9]:
for row in gplaydata:
    if row[0] == 'Quick PDF Scanner + OCR FREE':
        print(row)

['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']
['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']
['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80804', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']


In [10]:
for row in Appledata:
    if row[1] == 'Facebook':
        print(row)    

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


------------------------------------------------------------------------------------------------
As seen above the dataset holds some of the duplicate entries with minor differences in their entries. We see that in the above case that the index 3 of the entry list is different. This is the ratings column for the app.

We can further reason that the entry with most ratings must be the most recent one, and remove or delete the other entries sighting redundancy and comparitive outdatedness.

The duplicate entries can also be removed or avoided, through a different approach. This involves tackling the problem from the other feasible angle. That is, to simply move all other entries to a new list of lists except the duplicate entries.

Thus, we will proceed to create a new empty dictionary to store the maximum reviews of an app with its app name as the key. this way, we can easily move the required unique values into a new clean list.

# Create a dictionary to hold unique names and max reviews, call it max_reviews.


In [11]:
max_reviews = {}
for row in gplaydata:
    name = row[0] #Assign the app name to variable `name`
    n_reviews = float(row[3]) #Assign the numbers of reviews column to `n_reviews`
    if name in max_reviews and max_reviews[name] < n_reviews: #if the var name is present as key in dict max_reviews
        #and if value of the key name is less than the new number of reviews if and only then assign the new
        # more number of reviews to as the new value to the dict with key name.
        max_reviews[name] = n_reviews
    if name not in max_reviews: #if var name is not present as key in dict max_reviews, create a new key-value
        #pair with the number of reviews as the value(which may or may not be overwritten).
        max_reviews[name] = n_reviews


-----------------------------------------------------------
The below code would not make any sense, as there are no duplicates in the apple dataset.
-----------------------------------------------------------

In [12]:
android_clean = []
already_added = []
for row in gplaydata:
    #name = row[0]
    #n_reviews = float(row[3])
    if float(row[3]) == max_reviews[row[0]] and row[0] not in already_added:
        android_clean.append(row)
        already_added.append(row[0])

After creating a dict with the highest number of reviews for each app name (without any duplicates, as dicts cannot have duplicates). We can remove the duplicates entries from the header-less complete dataset. To do this, we first create two empty lists, namely, the android_clean and already_added. The former will hold the new cleaned dataset with no duplicate entry and the latter will be used to keep track of the process.

Using a for loop we run a row by row iteration through the gplaydata(without the header list). In the loop we add an condition that appends the android_clean list only when the number of reviews is same as that present inside the dict. for that app name and when the app name is not in the already_added list.
This way only the highest number of reviews and unique entries are added to the android_clean to make it a new list of lists. Post which, the already_added is also appended with the lastest app name for the iteration thus keeping track of duplicity.

In [13]:
print(len(android_clean))
print(android_clean[4])

9659
['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


# To check if any non-english character app names are present

In [14]:
def is_english(str):
    count = 0
    for i in str:
        if ord(i) > 127: # We know that non-eng char have ascii value greater than 127.
            count += 1
            if count > 3: # if there are more than 3 non-english char in the app names, we can than consider
                # the app name to be non-english, for simplicity and convenience.
                return False               
    return True

In [15]:
lis = ['Instagram', '爱奇艺PPS -《欢乐颂2》电视剧热播', 'Docs To Go™ Free Office Suite','Instachat 😜']
for i in lis:
    print(is_english(i))

True
False
True
True


We can now, use the above filter function to filter our dataset further to only include app name with english characters.

In [16]:
print(android_clean[4][0])

Paper flowers instructions


In [17]:
android_english = []
for row in android_clean:
    name = row[0]
    if is_english(name):
        android_english.append(row)
print(android_english[:4])

[['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up'], ['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up'], ['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']]


In [18]:
apl_english = []
for row in Appledata:
    name_apl = row[1]
    if is_english(name_apl):
        apl_english.append(row)
print(apl_english[:4])

[['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1'], ['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1'], ['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']]


In [19]:
android_english[0:3]

[['Photo Editor & Candy Camera & Grid & ScrapBook',
  'ART_AND_DESIGN',
  '4.1',
  '159',
  '19M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'January 7, 2018',
  '1.0.0',
  '4.0.3 and up'],
 ['U Launcher Lite – FREE Live Cool Themes, Hide Apps',
  'ART_AND_DESIGN',
  '4.7',
  '87510',
  '8.7M',
  '5,000,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'August 1, 2018',
  '1.2.4',
  '4.0.3 and up'],
 ['Sketch - Draw & Paint',
  'ART_AND_DESIGN',
  '4.5',
  '215644',
  '25M',
  '50,000,000+',
  'Free',
  '0',
  'Teen',
  'Art & Design',
  'June 8, 2018',
  'Varies with device',
  '4.2 and up']]

Thus, as shown above we see that both the datasets have been processed to have only app names with accordance to the ascii character set for English.

# Isolating free apps from others, as the main aim of the analysis is to find profitable free apps category.

In [20]:
android_free = []
for row in android_english:
    if row[-7] == 'Free':
        android_free.append(row)
print(android_free[:3])
print('\n')

apl_free = []
for row in apl_english:
    if row[4] == '0.0':
        apl_free.append(row)
#print(apl_english[:3])
print(apl_free[-6:-3])

[['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up'], ['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']]


[['1077585444', '針の穴', '15624192', 'USD', '0.0', '0', '0', '0.0', '0.0', '1.0.1', '4+', 'Games', '38', '5', '2', '1'], ['1065550288', 'Tchapper Messenger - Keep it private', '82187264', 'USD', '0.0', '0', '0', '0.0', '0.0', '2.5', '4+', 'Lifestyle', '37', '0', '9', '1'], ['1168960919', 'Golden Lotus Slots', '111412224', 'USD', '0.0', '0', '0', '0.0', '0.0', '1.1.0', '17+', 'Games', '37', '5', '3', '1']]


As seen above we have isolated both the free apps from both the datasets. For the apple dataset, the string '0.0' indicates the price of the app.

# Finding the most common app genres from both datasets

We will further proceed to find the most common app genres, as our business statergy focuses on building free apps, where the in-app ads from the apps will be the main source of income. With this idea in mind, it would be profitable to build apps in the genres, that are downloaded frequently by users from both Google Play and App store.

In [21]:
print(gplaycol)
print(Applecol)

[['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']]
[['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']]


In [22]:
# The above can also be done with a help of a function.

def freq_table(dataset, index):
    frq_tab = {}
    for row in dataset:
        if row[index] not in frq_tab:
            frq_tab[row[index]] = 0
        else:
            frq_tab[row[index]] += 1
    for key in frq_tab:  
        frq_tab[key] = round(((frq_tab[key]/len(dataset))*100), 2)
        
        # above code was simplified from the below code.
        '''
        frq_tab[key] /= len(dataset)
        frq_tab[key] *= 100 
        frq_tab[key] = round(frq_tab[key], 2)
        '''
        
    return frq_tab

plycatdict = freq_table(android_free, 1)
print(plycatdict, '\n')

appgendict = freq_table(apl_free, -5)
print(appgendict)

{'ART_AND_DESIGN': 0.63, 'AUTO_AND_VEHICLES': 0.91, 'BEAUTY': 0.59, 'BOOKS_AND_REFERENCE': 2.13, 'BUSINESS': 4.58, 'COMICS': 0.61, 'COMMUNICATION': 3.23, 'DATING': 1.85, 'EDUCATION': 1.15, 'ENTERTAINMENT': 0.95, 'EVENTS': 0.7, 'FINANCE': 3.69, 'FOOD_AND_DRINK': 1.23, 'HEALTH_AND_FITNESS': 3.07, 'HOUSE_AND_HOME': 0.81, 'LIBRARIES_AND_DEMO': 0.93, 'LIFESTYLE': 3.89, 'GAME': 9.71, 'FAMILY': 18.89, 'MEDICAL': 3.52, 'SOCIAL': 2.65, 'SHOPPING': 2.23, 'PHOTOGRAPHY': 2.93, 'SPORTS': 3.38, 'TRAVEL_AND_LOCAL': 2.32, 'TOOLS': 8.45, 'PERSONALIZATION': 3.31, 'PRODUCTIVITY': 3.88, 'PARENTING': 0.64, 'WEATHER': 0.79, 'VIDEO_PLAYERS': 1.78, 'NEWS_AND_MAGAZINES': 2.79, 'MAPS_AND_NAVIGATION': 1.39} 

{'Social Networking': 3.26, 'Photo & Video': 4.93, 'Games': 58.13, 'Music': 2.02, 'Reference': 0.53, 'Health & Fitness': 1.99, 'Weather': 0.84, 'Utilities': 2.48, 'Travel': 1.21, 'Shopping': 2.58, 'News': 1.3, 'Navigation': 0.16, 'Lifestyle': 1.55, 'Entertainment': 7.85, 'Food & Drink': 0.78, 'Sports': 2.11

In [23]:
def display_table(dataset, index):
    table = freq_table(dataset, index) # table is a dict. that needs to be sorted/ordered.
    table_display = [] # This list will be used to sort the contents into either ascending or descending.
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

The above function will show us the app category in a descending or ascending order by sorting through a list of tuples, that were created with a for loop on a dict. The sorted function does work on dict. as well , but only sorts the keys. Dict. are also considered as iterable as lists, tuples as shown and utilised in the code above. 

In [24]:
display_table(android_free, 1)
print('\n')
display_table(apl_free, -5)

FAMILY : 18.89
GAME : 9.71
TOOLS : 8.45
BUSINESS : 4.58
LIFESTYLE : 3.89
PRODUCTIVITY : 3.88
FINANCE : 3.69
MEDICAL : 3.52
SPORTS : 3.38
PERSONALIZATION : 3.31
COMMUNICATION : 3.23
HEALTH_AND_FITNESS : 3.07
PHOTOGRAPHY : 2.93
NEWS_AND_MAGAZINES : 2.79
SOCIAL : 2.65
TRAVEL_AND_LOCAL : 2.32
SHOPPING : 2.23
BOOKS_AND_REFERENCE : 2.13
DATING : 1.85
VIDEO_PLAYERS : 1.78
MAPS_AND_NAVIGATION : 1.39
FOOD_AND_DRINK : 1.23
EDUCATION : 1.15
ENTERTAINMENT : 0.95
LIBRARIES_AND_DEMO : 0.93
AUTO_AND_VEHICLES : 0.91
HOUSE_AND_HOME : 0.81
WEATHER : 0.79
EVENTS : 0.7
PARENTING : 0.64
ART_AND_DESIGN : 0.63
COMICS : 0.61
BEAUTY : 0.59


Games : 58.13
Entertainment : 7.85
Photo & Video : 4.93
Education : 3.63
Social Networking : 3.26
Shopping : 2.58
Utilities : 2.48
Sports : 2.11
Music : 2.02
Health & Fitness : 1.99
Productivity : 1.71
Lifestyle : 1.55
News : 1.3
Travel : 1.21
Finance : 1.09
Weather : 0.84
Food & Drink : 0.78
Reference : 0.53
Business : 0.5
Book : 0.4
Navigation : 0.16
Medical : 0.16
Catal

# Important to note
 It is crucial to understand at this point here, that the number of free apps for a category, does not necessarily have to mean, there are many number of users for that category. The descending list above only shows the number of such apps available as free in the app markets. However, the number of installs of an said app would indicate that the number of users or audience of that app or app category.


In [25]:
appgendict = freq_table(apl_free, -5)

for genre in appgendict:
    total = 0
    len_genre = 0
    for row in apl_free:
        genre_app = row[-5]
        if genre_app == genre:
            n_ratings = float(row[5])
            total += n_ratings        
            len_genre += 1
    avg_ratngs = total/len_genre
    print(genre,':',avg_ratngs)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


As seen above app genres like social networking, with navigation having the highest, than comes reference seem to have more than many reviews. They might have data that is more publicly reviewed than for an usual app category or genre. We can take a closer look to check such outliers.

In [26]:
for app in apl_free:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5




The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

Reference apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:



**However, *Reference apps* do not have lot of big player contribution and is relatively easy to build as it could not require any specific domain knowledge like other popular app genres like business, finance, etc.
An reference app would be required for users to refer to something they find curious while reading books or trying to learn a new topic of interest, the users can use reference apps to find meaning of new words, like a dictionary apps for example. An app in the *book category* would also be a good business proposition in the App store.**

# Most Popular Apps by Genre on Google Play

In [27]:
plycatdict = freq_table(android_free, 1)

for cat in plycatdict: # first loop will go through each cat. in the cat. dict. one by one to -                         
    total_per_genre = 0 # -sum up all the installs
    len_genre = 0
    for row in android_free: 
        if row[1] == cat:
           # n_installs = row[5] # saves the number of installs
           # n_installs = n_installs.replace('+','') 
           # n_installs = float(n_installs.replace(',',''))
            total_per_genre += float(row[5].replace('+','').replace(',','')) #The str is replaced of commas -
            len_genre += 1 # -and + signs and converted to a float value.
    avg_installs_per_genre = total_per_genre/len_genre
    print(cat,':',avg_installs_per_genre)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3697848.1731343283
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

We see 'ART_AND_DESIGN' category as the highest number of installs. Let us further look at the apps of the category.

In [28]:
for row in android_free:
    if row[1] == 'ART_AND_DESIGN':
        print(row[0], ":", row[5])

Photo Editor & Candy Camera & Grid & ScrapBook : 10,000+
U Launcher Lite – FREE Live Cool Themes, Hide Apps : 5,000,000+
Sketch - Draw & Paint : 50,000,000+
Pixel Draw - Number Art Coloring Book : 100,000+
Paper flowers instructions : 50,000+
Smoke Effect Photo Maker - Smoke Editor : 50,000+
Infinite Painter : 1,000,000+
Garden Coloring Book : 1,000,000+
Kids Paint Free - Drawing Fun : 10,000+
Text on Photo - Fonteee : 1,000,000+
Name Art Photo Editor - Focus n Filters : 1,000,000+
Tattoo Name On My Photo Editor : 10,000,000+
Mandala Coloring Book : 100,000+
3D Color Pixel by Number - Sandbox Art Coloring : 100,000+
Learn To Draw Kawaii Characters : 5,000+
Photo Designer - Write your name with shapes : 500,000+
350 Diy Room Decor Ideas : 10,000+
FlipaClip - Cartoon animation : 5,000,000+
ibis Paint X : 10,000,000+
Logo Maker - Small Business : 100,000+
Boys Photo Editor - Six Pack & Men's Suit : 100,000+
Superheroes Wallpapers | 4K Backgrounds : 500,000+
HD Mickey Minnie Wallpapers : 5

# Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

# To analyse the Genres column of google play dataset to find any similarities between the category column.

In [37]:
plygendict = freq_table(android_free, -5)
print(plygendict)

for gen in plygendict:
    total_instl = 0
    len_app = 0
    for row in android_free:
        if row[-5] == gen:
            total_instl += float(row[5].replace('+','').replace(',',''))
            len_app += 1
    avg_instl = total_instl/len_app
    print(gen,":",avg_instl)
print(gplaycol)

{'Everyone': 81.43, 'Teen': 11.05, 'Everyone 10+': 3.25, 'Mature 17+': 4.17, 'Adults only 18+': 0.02, 'Unrated': 0.01}
Everyone : 7251919.606400665
Teen : 16812875.982653063
Everyone 10+ : 13846928.356401384
Mature 17+ : 6539796.113207547
Adults only 18+ : 666666.6666666666
Unrated : 25250.0
[['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']]


In [38]:
for row in android_free:
    if row[-5] == 'Teen':
        print(row[0],":", row[5])

Sketch - Draw & Paint : 50,000,000+
Tattoo Name On My Photo Editor : 10,000,000+
I Creative Idea : 10,000+
Android Auto - Maps, Media, Messaging & Voice : 10,000,000+
Selfie Camera Photo Editor & Filter & Sticker : 50,000+
Skin Care and Natural Beauty : 100,000+
Book store : 1,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
Litnet - E-books : 100,000+
GANMA! - All original stories free of charge for all original comics : 1,000,000+
Manga Rock - Best Manga Reader : 1,000,000+
Manga - read Thai translation : 10,000+
Comic Es - Shojo manga / love comics free of charge ♪ ♪ : 100,000+
comico Popular Original Cartoon Updated Everyday Comico : 5,000,000+
Manga Zero - Japanese cartoon and comic reader : 1,000,000+
Marvel Unlimited : 1,000,000+
Tapas – Comics, Novels, and Stories : 1,000,000+
Manga Mania - Best online manga reader : 10,000+
Best Wallpapers Backgrounds(100,000+ 4K HD) : 10,000+
Messenger for SMS : 10,000,000+
Omlet Chat : 10,00

There seems to be lot of apps build for teen around education, entertainment, etc. This category can be used for our business idea and a suitable app that falls under our interest of business policy 